test/ex_plasma/client_test.exs

```diff
@@ -61,8 +61,11 @@ defmodule ExPlasma.ClientTest do
   describe "submit_block/3" do
     test "it submits a block of transactions" do
       use_cassette "submit_block", match_requests_on: [:request_body] do
+        input = %Utxo{blknum: 0, txindex: 0, oindex: 0}
+        output = %Utxo{amount: 0, currency: <<0::160>>, owner: <<0::160>>}
```

Comment:
if you omit `amount` from output building you get a crash
```
1) test submit_block/3 it submits a block of transactions (ExPlasma.ClientTest)
     test/ex_plasma/client_test.exs:62
     ** (FunctionClauseError) no function clause matching in ExPlasma.Utxo.to_output_rlp/1

     The following arguments were given to ExPlasma.Utxo.to_output_rlp/1:

         # 1
         %ExPlasma.Utxo{amount: nil, blknum: nil, currency: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, oindex: nil, output_type: 1, owner: <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>, txindex: nil}

     Attempted function clauses (showing 4 out of 4):

         def to_output_rlp(%{amount: amount} = utxo) when is_integer(amount)
         def to_output_rlp(%{currency: <<_::integer()-size(336)>> = currency} = utxo)
         def to_output_rlp(%{owner: <<_::integer()-size(336)>> = owner} = utxo)
         def to_output_rlp(%{currency: <<_::integer()-size(160)>>, owner: <<_::integer()-size(160)>>, amount: <<_::integer()-size(256)>>} = utxo)

     code: |> Block.new()
     stacktrace:
       (ex_plasma) lib/ex_plasma/utxo.ex:218: ExPlasma.Utxo.to_output_rlp/1
       (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
       (ex_plasma) lib/ex_plasma/transaction.ex:140: ExPlasma.Transaction.to_rlp/1
       (ex_plasma) lib/ex_plasma/transaction.ex:166: ExPlasma.Transaction.encode/1
       (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2
       (ex_plasma) lib/ex_plasma/block.ex:44: ExPlasma.Block.merkle_root_hash/1
       (ex_plasma) lib/ex_plasma/block.ex:39: ExPlasma.Block.new/1
       test/ex_plasma/client_test.exs:70: (test)
```

is this expected or would we handle this specifically so that developer knows what needs to be populated for outputs (and inputs).

Reply:
I've added checks for this in  6c32c79, but will open a follow up PR to validate the struct before any conversions from method like `to_output_rlp/1` or even `to_input_rlp/1`. Nice catch 🙏

:clipboard: https://github.com/omisego/ex_plasma/issues/28